src/Lean/Parser/Do.lean

```diff
@@ -8,23 +8,23 @@ import Lean.Parser.Term
 namespace Lean
 namespace Parser
 
-builtin_initialize registerBuiltinParserAttribute `builtinDoElemParser ``Category.doElem
-builtin_initialize registerBuiltinDynamicParserAttribute `doElemParser `doElem
+builtin_initialize registerBuiltinParserAttribute `builtin_doElem_parser ``Category.doElem
+builtin_initialize registerBuiltinDynamicParserAttribute `doElem_parser `doElem
```

Comment:
The reason this attribute is not completely snake cased (although I suppose it looks fine as is) is because `elab_rules` and `macro_rules` make the assumption that you can just stick `Parser` (now `_parser`) at the end of the category name to derive the parser name. So this function is somewhat misleading and should probably not have the first argument at all, to avoid the implication that it can be changed.